Temperature_API/crud.py

```diff
@@ -0,0 +1,89 @@
+from datetime import datetime
+
+import httpx
+from sqlalchemy.future import select
+from sqlalchemy.ext.asyncio import AsyncSession
+from City_CRUD_API.models import City
+from Temperature_API import models
+from Temperature_API.models import Temperature
+from Temperature_API.schemas import TemperatureCreate
+
+
+WEATHER_API_URL = "http://api.weatherapi.com/v1/current.json"
+API_KEY = "d0eae92abc5b4efaa76144848242007"
+
+
+async def get_all_temperatures(db: AsyncSession):
+    result = await db.execute(select(models.Temperature))
+    return result.scalars().all()
+
+
+async def get_temperature_by_city(db: AsyncSession, city_id: int):
+    result = await db.execute(
+        select(models.Temperature).
+        filter(models.Temperature.city_id == city_id)
+    )
+    return result.scalars().first()
+
+
+async def update_temperatures(db: AsyncSession):
+    # Get list of all cities
+    result = await db.execute(select(City))
+    cities = result.scalars().all()
```

Comment:
Ensure that the `City` model is correctly imported and available in the codebase to avoid runtime errors when fetching all cities.